examples/account_storage/DiskHungry.artifacts/approval.teal

```diff
@@ -89,203 +90,195 @@ retsub
 
 // flip_bit
 flipbit_2:
-store 11
-store 10
-load 10
+proto 2 0
+frame_dig -2
 txnas Accounts
-load 11
+frame_dig -1
 intc_3 // 8
 /
 callsub impl_6
-store 12
-load 12
-load 11
+store 6
+load 6
+frame_dig -1
 intc_3 // 8
 %
-load 12
-load 11
+load 6
+frame_dig -1
 intc_3 // 8
 %
 getbit
 !
 setbit
-store 12
-load 10
+store 6
+frame_dig -2
 txnas Accounts
-load 11
+frame_dig -1
 intc_3 // 8
 /
-load 12
+load 6
 callsub impl_7
 retsub
 
 // encode_uvarint_impl
 encodeuvarintimpl_3:
-store 8
-store 7
-load 8
-load 7
+proto 2 1
+frame_dig -1
+frame_dig -2
 intc 4 // 128
 >=
 bnz encodeuvarintimpl_3_l2
-load 7
+frame_dig -2
 intc 5 // 255
 &
 itob
 extract 7 1
 b encodeuvarintimpl_3_l3
 encodeuvarintimpl_3_l2:
-load 7
+frame_dig -2
 pushint 7 // 7
 shr
-load 7
+frame_dig -2
 intc 5 // 255
 &
 intc 4 // 128
 |
 itob
 extract 7 1
-load 7
-load 8
-uncover 3
-uncover 3
 callsub encodeuvarintimpl_3
-cover 2
-store 8
-store 7
 encodeuvarintimpl_3_l3:
 concat
 retsub
 
 // populate_template_program
 populatetemplateprogram_4:
+proto 0 1
 intc_1 // 0
-store 3
+store 2
 intc_1 // 0
+store 3
+bytec_0 // ""
 store 4
 bytec_0 // ""
 store 5
-bytec_0 // ""
-store 6
-load 2
+frame_dig -1
```

Comment:
this is the problematic place, where `populate_template_program` should specify num of args in subroutine related proto, or generated code will have something wrong...

Reply:
Simply removing the subroutine decorator on the inner method fixes the issue here